app/test/unit/endpoint-capabilities-test.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  endpointSatisfies,
+  VersionConstraint,
+} from '../../src/lib/endpoint-capabilities'
+import { SemVer, parse } from 'semver'
+import { getDotComAPIEndpoint } from '../../src/lib/api'
+import { forceUnwrap } from '../../src/lib/fatal-error'
+
+describe('endpoint-capabilities', () => {
+  describe('endpointSatisfies', () => {
+    it('recognizes github.com', () => {
+      expect(testDotCom(true)).toBeTrue()
+      expect(testDotCom(false)).toBeFalse()
+    })
+
+    it('recognizes GHES', () => {
+      expect(testGHES(false)).toBeFalse()
+      expect(testGHES(true)).toBeTrue()
+    })
+
+    it('recognizes GHAE', () => {
+      expect(testGHAE(false)).toBeFalse()
+      expect(testGHAE(true)).toBeTrue()
+    })
+
+    it('assumes GHAE versions', () => {
+      expect(testGHAE('>= 3.3.1')).toBeFalse()
+      expect(testGHAE('>= 3.3.0')).toBeTrue()
+    })
+
+    it('assumes GHES versions', () => {
+      expect(testGHES('>= 3.1.1')).toBeFalse()
+      expect(testGHES('>= 3.1.0')).toBeTrue()
+    })
```

Comment:
Maybe it's worth adding a comment explaining what do they mean about "assuming the GHAE/GHES versions", where those constants live, and mentioning that these tests will need to be updated when those constants change.